src/main/java/io/fabric8/maven/docker/util/WaitUtil.java

```diff
@@ -48,6 +48,26 @@
 
     private WaitUtil() {}
 
+    public static void wait(int wait, Callable<Void> callable) throws ExecutionException {
+        try {
+            FutureTask<Void> task = new FutureTask<>(callable);
+            task.run();
+
+            /*
+             * pad the wait time so that the future will never time out but still complete - if pad == 0,
+             * set to default docker timeout
+             */
+            task.get(((wait == 0) ? 10 : wait) + 5, TimeUnit.SECONDS);
+        } catch (@SuppressWarnings("unused") InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } catch (@SuppressWarnings("unused") TimeoutException e) {
+            /*
+             * this should not happen b/c the wait time has been padded such that if the docker daemon fails
+             * to kill the container in the alloted time, it will be booted out via the ExecutionException
+             */
+        }
```

Comment:
i'm ok w/ a log statement but i still need to figure out the best way to deal w/ that given this class is static and i don't really want to pass it in as an argument. 

maybe i'll just throw the exception and have the caller deal w/ it b/c the logger will be available there.